py-rse/bash-basics.Rmd

```diff
@@ -1315,13 +1330,16 @@ The vertical bar `|` between the `wc` and `sort` commands
 tells the shell that we want to use the output of the command on the left
 as the input to the command on the right.
 Putting it another way,
-the downstream command doesn't read from a file;
-instead,
+the downstream command doesn't read from a file.
+Instead,
 it reads the output of the upstream command.
 
+FIXME: diagram of pipes
+
 We can use `|` to build pipes of any length.
 For example,
-we can use the command `head` to get just the first three lines of sorted data, the three shortest books:
+we can use the command `head` to get just the first three lines of sorted data,
+which shows us the three shortest books:
 
 ```shell
 $ wc -l *.txt | sort -n | head -n 3
```

Comment:
This is the first time our learners see an option that also requires an argument.  I.e. given the earlier discussion about options and arguments, is it worth pointing out that the `3` is actually being given to the `-n` option, not as input to `head`.

Reply:
Good catch - paragraph added.